### ISSUE TITLE
Update tasks to block dispatch

### ISSUE THREAD

**Issue by hschallhorn** [department-of-veterans-affairs/caseflow#14057]
The following tasks should be updated to block dispatch:
- CongressionalInterestMailTask
- FoiaRequestMailTask
- PrivacyActRequestMailTask
- PrivacyComplaintMailTask
- FoiaColocatedTask
- FoiaTask
- PrivacyActTask
- ChangeHearingDispositionTask

### Acceptance Criteria
- [ ] The tasks above block dispatch

**Comment by yoomlam** [comment]
Updating prod data: these tasks should not block distribution.

**Comment by hschallhorn** [comment]
@araposo-tistatech Can we get some clarification around how the board envisions `ChangeHearingDispositionTask` would not block distributino? This task allows users to mark a hearing as 
1. [Postponed](https://github.com/department-of-veterans-affairs/caseflow/blob/ee6a5a7876e952bf9b453e0bb376b0e3482fb8a8/app/models/tasks/assign_hearing_disposition_task.rb#L98-L104) (a new hearing will be scheduled, which also blocks distribution)
1. [No show](https://github.com/department-of-veterans-affairs/caseflow/blob/ee6a5a7876e952bf9b453e0bb376b0e3482fb8a8/app/models/tasks/assign_hearing_disposition_task.rb#L106-L112) ([appeal is put on hold waiting for the veteran to send in mail as to why they they did not show](https://github.com/department-of-veterans-affairs/caseflow/blob/7adbc4bdb0e06b0741e9576d0c682a4a40ae6b92/app/models/tasks/no_show_hearing_task.rb#L8-L18), also blocks distribution)
1. [Cancelled](https://github.com/department-of-veterans-affairs/caseflow/blob/ee6a5a7876e952bf9b453e0bb376b0e3482fb8a8/app/models/tasks/assign_hearing_disposition_task.rb#L82-L96) (an evidence submission window is opened which also blocks dispatch)
1. [Held](https://github.com/department-of-veterans-affairs/caseflow/blob/ee6a5a7876e952bf9b453e0bb376b0e3482fb8a8/app/models/tasks/assign_hearing_disposition_task.rb#L114-L124) (an evidence submission window is opened which also blocks distribution)

In my mind, this task has to block distribution as the appeal cannot continue forward with distribution if this task is not completed

**Comment by araposo-tistatech** [comment]
@hschallhorn I've sent an email to the Board.

**Comment by lomky** [comment]
We have some circular dependencies in our tickets to write the FE/BE/New behavior - I've noted a place in the BE implementation that this ticket should update for the health of the test suite:

https://github.com/department-of-veterans-affairs/caseflow/pull/14957/files#diff-e266c6cdf27b55f03934aa7e12a1c35bR54-R56

**Comment by araposo-tistatech** [comment]
@hschallhorn Mike S. at the Board stated the ChangeHearingDispositionTask should block dispatch.

**Comment by hschallhorn** [comment]
_[what is this chart?](https://github.com/department-of-veterans-affairs/caseflow/issues/11533)_
```bash
1 | 
2 | ||
3 | |||||
5 | ||
8 | 
```

Will have to be careful with child tasks. (FoiaTasks are always children of FoiaColocatedTasks) Don't disrupt the task subtree!

- [ ] Tasks that _should_ block dispatch that are currently blocking distribution tasks should be moved to blocking dispatch
  - [ ] Update distribution task if moving this task makes the appeal ready to be distributed
- [ ] Tasks that _should_ block dispatch that are currently blocking a judge assign or decision review tree (via attorney task or possibly via a colocated task) should be moved to blocking dispatch
  - [ ] Update judge or attorney or colocated task if moving this task makes the task ready to be worked

#### Why 2?
* Optimism!

#### Why 3?
* Messing around in prod always a little iffy
* Will require a second set of eyes for prod

#### Why 5?
* Could run into special cases we did not consider
* Hard to fully vet in dev

**Comment by lomky** [comment]
[Researching](https://hackmd.io/zLVLVNCURfWYcBD6nqo9Mg)

**Comment by lomky** [comment]
Placing back into the team queue. Discovery work is done for Mail Tasks and in-progress for Foia Tasks (see hackMD). Completion blocked by #14056 with is awaiting gov partner response
